aea/helpers/exec_timeout.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Python code execution time limit tools."""
+
+import asyncio
+import ctypes
+import signal
+import threading
+from abc import ABC, abstractmethod
+from asyncio import Future
+from asyncio.events import AbstractEventLoop
+from types import TracebackType
+from typing import Type
+
+
+class TimeoutResult:
+    """ Result of ExecTimeout context manager """
+
+    def __init__(self):
+        self.cancelled_by_timeout = False
```

Comment:
could be made private (minor point though)

Reply:
done